### ISSUE TITLE
@Unique decorator don't recognize column names

### ISSUE THREAD

**Issue by webleaf** [typeorm/typeorm#3732]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[x] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

If column name is specified via `@Column({name: string})` or `@JoinColumn({name: string})`, such column don't recognized by `@Unique` and we still should use Entity original property name.

```typescript
@Entity()
@Unique(["field_name_1", "field_name_2"])
export class EntityName {

    @PrimaryGeneratedColumn()
    public id: number;

    @ManyToOne(type => OtherEntityName)
    @JoinColumn({name: "field_name_1"})
    public field1: OtherEntityName;

    @Column({name: "field_name_2"})
    public field2: string;
}
```
```
Error: Unique constraint contains column that is missing in the entity: field_name_1
```

**Comment by vlapo** [comment]
Hey @webleaf. You have to use entity field name not a DB column name:
```typescript
@Unique(["field1", "field2"])
```

**Comment by webleaf** [comment]
@vlapo yes, I know. Then it's docs bug:

> This decorator allows you to create a database unique constraint **for a specific column or columns**.

**Comment by vlapo** [comment]
Feel free to contribute :) Note in https://typeorm.io/#/indices should be useful.

**Comment by sheychen290** [comment]
Maybe related...
```
@Entity()
@Unique(["field1", "field2"])
export class EntityName {
    @ManyToOne(type => OtherEntityName, {primary: true})
    @JoinColumn({name: "field_name_1"})
    public field1: OtherEntityName;

    @Column({name: "field_name_2", primary: true})
    public field2: string;
}
```
Produce ```Error: Index contains column that is missing in the entity: field1```
Note: Same problem with ```field_name_1```

**Comment by pleerock** [comment]
fixed by #4233

**Comment by qathom** [comment]
Hello,

I'd like to share my approach.
Hope it can help someone.

```js
function notUnique<Entity>(entityA: Entity, entityB: Entity, tableUnique: (keyof Entity)[]): boolean {
  return tableUnique.reduce((acc, key) => {
    if (entityA[key] === entityB[key]) {
      acc += 1;
    }
    return acc;
  }, 0) === tableUnique.length;
}

/**
 * 
 * @param repository - The entity repository
 * @param entities - The list of entities to save
 * @param tableUnique - The unique constraint argument such as @Unique(['field1', 'field2'])
 */
export async function saveAndIgnoreOnDuplicate<Entity>(repository: Repository<Entity>, entities: Entity[], tableUnique: (keyof Entity)[] = []): Promise<Entity[]> {
  try {
    const dbEntities = await repository.find();
    const pushEntities: Entity[] = entities
      // [1] Database duplicates
      .filter((entity) => dbEntities.findIndex(dbEntity => notUnique(entity, dbEntity, tableUnique)) === -1)
      // [2] Same transaction duplicates
      .filter((entity, index, self) => self.findIndex((item) => notUnique(item, entity, tableUnique)) === index);

    return await repository.save(pushEntities);
  } catch (err) {
    if (err.code !== 'ER_DUP_ENTRY') {
      // Throw exception only if it is something else than duplicate entry
      throw err;
    }
  }

  return [];
}
```
